LIS3DH.class.nut

```diff
@@ -0,0 +1,362 @@
+    function configureClickInterrupt(state, clickType = null, threshold = 1.1, timeLimit = 5, latency = 10, window = 50) {
+        // Set default value for clickType (since we're using statics we can't set in function definition)
+        if (clickType == null) clickType = SINGLE_CLICK;
+
+        // Set the enable / disable flag
+        _setRegBit(CTRL_REG3, 7, state ? 1 : 0);
+
+        // If they disabled the click interrupt, set click_cfg register and return
+        if (!state) {
+            _setReg(CLICK_CFG, 0x00);
+            return;
+        }
+
+        // Set the CLICK_CFG register
+        _setReg(CLICK_CFG, clickType);
+
+        // Set the CLICK_THS register
+        if (threshold < 0) { threshold = threshold * -1.0; }    // Make sure we have a positive value
+        if (threshold > _range) { threshold = range; }          // Make sure it doesn't exceed the _range
+
+        threshold = (((threshold * 1.0) / (_range * 1.0)) * 127).tointeger();
+        _setReg(CLICK_THS, threshold);
+
+        // Set the TIME_LIMIT register (max time for a click)
+        _setReg(TIME_LIMIT, timeLimit);
+        // Set the TIME_LATENCY register (min time between clicks for double click)
+        _setReg(TIME_LATENCY, latency);
+        // Set the TIME_WINDOW register (max time for double click)
+        _setReg(TIME_WINDOW, window);
+    }
+
+    // Enable/Disable Data Ready Interrupt 1 on Interrupt Pin
+    function configureDataReadyInterrupt(state) {
+        _setRegBit(CTRL_REG3, 4, state ? 1 : 0);
+    }
+
+    // Enables/disables interrupt latching
+    function setInterruptLatching(state) {
+        _setRegBit(CTRL_REG5, 1, state ? 1 : 0);
+    }
+
+    // Returns interrupt registers as a table, and clears the INT1_SRC register
+    function getInterruptTable() {
+        local int1 = _getReg(INT1_SRC);
+        local click = _getReg(CLICK_SRC);
+
+        return {
+            "int1":         (int1 & 0x40) != 0,
+            "xLow":         (int1 & 0x01) != 0,
+            "xHigh":        (int1 & 0x02) != 0,
+            "yLow":         (int1 & 0x04) != 0,
+            "yHigh":        (int1 & 0x08) != 0,
+            "zLow":         (int1 & 0x10) != 0,
+            "zHigh":        (int1 & 0x20) != 0,
+            "click":        (click & 0x40) != 0,
+            "singleClick":  (click & 0x10) != 0,
+            "doubleClick":  (click & 0x20) != 0
+        }
+    }
+
+
+    //-------------------- PRIVATE METHODS --------------------//
+    function _twosComp(value, mask) {
+        value = ~(value & mask) + 1;
+        return value & mask;
+    }
+
+    function _getReg(reg) {
+        local val = _i2c.read(_addr, format("%c", reg), 1);
+        if (val != null) {
+            return val[0];
+        } else {
+            return null;
+        }
+    }
+
+    function _setReg(reg, val) {
+        _i2c.write(_addr, format("%c%c", reg, (val & 0xff)));
+    }
+
+    function _setRegBit(reg, bit, state) {
```

Comment:
_setRegBit sets a specific bit in the register (so state is the state we're setting of that bit).